### ISSUE TITLE
about mask

### ISSUE THREAD

**Issue by INTOUCHABLE-VS** [fzh0917/SparseTT#8]
I noticed that you used mask before calculating TopK in Figure 4, but in the training code attn_mask has always been None, so what is the role of mask?

**Comment by fzh0917** [comment]
the argument `attn_mask` is used to prevent attention to certain positions (see [the official docs](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html#torch.nn.MultiheadAttention.forward) for more details). However, the sparse multi-head attention mechanism in our work is not implemented by setting `attn_mask`. Thus, it has always been `None` in the training and testing phases. 
The key implementation of the sparse multi-head attention mechanism in our work is here: https://github.com/fzh0917/SparseTT/blob/791245c2317b6ccda38ea2aac44bce62879bfb06/videoanalyst/model/utils/multi_head_attention_topk.py#L208

If you have any questions about the code implementation, please feel free to ask me. :smile:

**Comment by INTOUCHABLE-VS** [comment]
First, thank you for your reply！I have no doubts about the key implementation of the sparse multi-head attention mechanism. I just saw that **figure 4** in your paper uses **mask** in calculating Sparse scaled dot-product self-attention, but not in training and testing. I have a little doubt about the role of mask.

**Comment by fzh0917** [comment]
Yes. In this work, `attn_mask` is always set to `None`, which means that similarity is calculated for all pixel pairs between `Query` and `Key`.

**Comment by INTOUCHABLE-VS** [comment]
OK, thank you very much for your reply
